### DISCUSSION TITLE
Getting Error While using filament table as dashboard widget

### DISCUSSION THREAD

**Discussion by titusfrezer** [filamentphp/filament#8575] [Help] [answered]
![image](https://github.com/filamentphp/filament/assets/53559440/22872ba4-81dd-4bc9-94ca-67f2e6725ac1)

I used the above code to join 2 tables and output the array using the table in filament but got the below error.

Filament\Widgets\TableWidget::getTableRecordKey(): Return value must be of type string, null returned;

**Comment by titusfrezer** [comment, marked as answer]
Never mind, the issue was what I was trying to select from my db(just selecting a name from the table) so filament was claiming me for that. I added ID too when I selected it from the table and the error was gone.

**Comment by renoanthus** [comment]
public function getTableRecordKey(Model $record): string
{
    return $record->uniqueColumn;
}

**Comment by ajayjoshioutdosolutions** [comment]
This worked for me
public function getTableRecordKey($record): string
    {
        return (string) $record->module_id; 
    }
